src/library_gl.js

```diff
@@ -5312,7 +5312,7 @@ var LibraryGL = {
         if (!GL.currArrayBuffer) {
           GLImmediate.firstVertex = end ? start : TOTAL_MEMORY; // if we don't know the start, set an invalid value and we will calculate it later from the indices
           GLImmediate.lastVertex = end ? end+1 : 0;
-          GLImmediate.vertexData = {{{ makeHEAPView('F32', 'GLImmediate.vertexPointer', '(end ? GLImmediate.vertexPointer + (end+1)*GLImmediate.stride : TOTAL_MEMORY)') }}}; // XXX assuming float
+          GLImmediate.vertexData = HEAPF32.subarray(GLImmediate.vertexPointer >> 2, end ? (GLImmediate.vertexPointer + (end+1)*GLImmediate.stride) >> 2 : undefined); // XXX assuming float
```

Comment:
Also, I'm not quite sure I understand how this works in split heap mode - in that mode, there is no HEAPF32, but that is replaced by a set of heaps in `HEAPF32s`? Also, is it possible that this kind of contiguous `.subarray()` access straddles two separate heap buffers?